MonoGame.Framework/Input/IImeService.cs

```diff
@@ -0,0 +1,67 @@
+// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using System;
+
+namespace Microsoft.Xna.Framework
+{
+    /// <summary>
+    /// Interface to handle text composition.
+    /// </summary>
+    public interface IImeService
+    {
+        /// <summary>
+        /// Enable the system IMM service to support composited character input.
+        /// This should be called when you expect text input from a user and you support languages
+        /// that require an IME (Input Method Editor).
+        /// </summary>
+        void StartTextInput();
+
+        /// <summary>
+        /// Stop the system IMM service.
+        /// </summary>
+        void StopTextInput();
+
+        /// <summary>
+        /// Show the Ime Candidate window renderer by the OS.
+        /// In SDL2 platform it's always true since there's no API to get candidate texts in SDL2.
+        /// In WindowsDX platform, it's possible to choose show candiate window renderer by the OS or render it yourself.
+        /// </summary>
+        bool ShowOSImeWindow { get; set; }
+
+        /// <summary>
+        /// Position Y of virtual keyboard, for mobile platforms has virtual keyboard.
+        /// </summary>
+        int VirtualKeyboardHeight { get; }
+
+        /// <summary>
+        /// Update the Ime service. For instance, stop the text input when clicking on the other
+        /// screen area than the virtual keyboard poped up.
+        /// </summary>
+        void Update(GameTime gameTime);
```

Comment:
We could update the `ImeService` internally so users don't have to. That would simplify usage a little.

Reply:
I agree this.

Reply:
@Jjagg If update `ImmService` internally, maybe make it a abstract class that could be better than an interface? Since interface cannot have non-public members.